sample-app/src/main/kotlin/com/sats/dna/sample/internal/SharedTransitions.kt

```diff
@@ -0,0 +1,24 @@
+@file:OptIn(ExperimentalSharedTransitionApi::class)
+
+package com.sats.dna.sample.internal
+
+import androidx.compose.animation.AnimatedContentScope
+import androidx.compose.animation.ExperimentalSharedTransitionApi
+import androidx.compose.animation.SharedTransitionScope
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.compositionLocalOf
+import androidx.compose.ui.Modifier
+
+@Composable
+internal fun Modifier.trySharedBounds(key: String): Modifier {
+    val sharedTransitionScope = LocalSharedTransitionScope.current ?: return this
+    val animatedContentScope = LocalAnimatedContentScope.current ?: return this
+
+    return with(sharedTransitionScope) {
+        val state = rememberSharedContentState(key)
+        sharedBounds(state, animatedContentScope)
+    }
+}
+
+internal val LocalAnimatedContentScope = compositionLocalOf<AnimatedContentScope?> { null }
+internal val LocalSharedTransitionScope = compositionLocalOf<SharedTransitionScope?> { null }
```

Comment:
This is probably not the most elegant solution, but it lets us at least apply the `Modifier` without needing to be in the context of a `SharedTransitionScope` (provided by the very top-level `SharedTransitionScope {}` composable) or having direct access to an `AnimatedContentScope` (provided in each screen's `NavGraphBuilder.composable {}` block).